### ISSUE TITLE
Activist Portal: Calendar modal trigger button lacks text label and other attributes

### ISSUE THREAD

**Issue by henrycatalinismith** [zetkin/app.zetkin.org#3640]
### The feature

In the activist portal there's a calendar icon button. Clicking it triggers a date picker modal. Here's a screenshot of it while focused in VoiceOver.

![Image](https://github.com/user-attachments/assets/3d962b97-3bfa-4129-9bb1-ffd0a9f8b258)

### The bug

VoiceOver's announcing it as `button` because the button lacks a text label. When a button contains only image content like in this case then aria-label should be used to provide a text label for assistive tech as per [1.1.1 Non-text Content](https://www.w3.org/WAI/WCAG21/Understanding/non-text-content.html).

Clicking the button triggers a modal dialog, but the button lacks the appropriate ARIA attributes exposing this aspect of its behavior to assistive tech. This falls under [4.1.2 Name Role Value](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html). It needs aria-haspopup="dialog" and also aria-expanded="false" (which should ideally switch to true when expanded).

## Steps to reproduce

1. Go to http://localhost:3000/o/1
2. Fire up VoiceOver
3. Focus the calendar modal trigger button highlighted in the screenshot above

### Expected Behaviour

VoiceOver should announce something like

> Select a date, dialog popup, button

### Actual Behaviour

It just says

> button
